### PR TITLE
add String[X] types

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/WeakRefStrings.jl/issues/85

CC @quinnj

### CODE REVIEW

**Comment by bkamins** [comment]
and @nalimilan

File: README.md [hunk 1]
```diff
@@ -33,6 +33,36 @@ Contributions are very welcome, as are feature requests and suggestions. Please
...
 
-Additional documentation is available at the REPL for `?WeakRefStringArray` and `?WeakRefString`.
+A set of custom string types of various fixed sizes. Each inline string is a
+custom primitive type and can benefit from being stack friendly by avoiding
+allocations/heap tracking in the GC. When used in an array, the elements are
+able to be stored inline since each one has a fixed size. Currently support
+inline strings from 1 byte up to 255 bytes.
+
+The following types are supported: `String1`, `String3`, `String7`, `String15`,
+`String31`, `String64`, `String127`, `String255`.
```
Comment (bkamins) [inline README.md:45]:
```suggestion
`String31`, `String63`, `String127`, `String255`.
```

**Comment by bkamins** [comment]
@quinnj - this should be merged after WeakRefStrings.jl 1.3.0 release is made.

**Review by quinnj — APPROVED** [review]
Thanks!

**Pull request merged**